src/plugins/Multipart.js

```diff
@@ -24,25 +29,28 @@ module.exports = class Multipart extends Plugin {
   }
 
   upload (file, current, total) {
+    const opts = Object.assign({}, this.opts, file.multipart || {})
+
     this.core.log(`uploading ${current} of ${total}`)
     return new Promise((resolve, reject) => {
       // turn file into an array so we can use bundle
-      // if (!this.opts.bundle) {
+      // if (!opts.bundle) {
       //   files = [files[current]]
       // }
 
       // for (let i in files) {
-      //   formPost.append(this.opts.fieldName, files[i])
+      //   formPost.append(opts.fieldName, files[i])
       // }
 
       const formPost = new FormData()
-      formPost.append(this.opts.fieldName, file.data)
 
-      Object.keys(file.meta).forEach((item) => {
-        // console.log(file.meta, file.meta[item])
+      const fields = Array.isArray(opts.fields) ? opts.fields : Object.keys(file.meta)
+      fields.forEach((item) => {
         formPost.append(item, file.meta[item])
       })
 
+      formPost.append(opts.fieldName, file.data)
```

Comment:
Moved this one to the end because S3 requires the `file` to be the last parameter … hopefully nothing else requires it to be first! :/